pkg/client/googlegroup_github/mapper.go

```diff
@@ -19,15 +19,88 @@ import (
 	"context"
 	"fmt"
 	"os"
+	"slices"
 	"strings"
 
 	"google.golang.org/protobuf/encoding/prototext"
 
 	"github.com/abcxyz/pkg/logging"
 	"github.com/abcxyz/team-link/apis/v1alpha3"
+	"github.com/abcxyz/team-link/pkg/github"
 	"github.com/abcxyz/team-link/pkg/groupsync"
 )
 
+// GroupMapper implements groupsync.OneToManyGroupMapper
+// For now it was be used as Mapper between GoogleGroup
+// and GitHubTeam.
+// Both GoogleGroupMapper and GitHubMapper have similar
+// variables and functions, this way we don't need to repeatly
+// write similar function for both on them.
+type GroupMapper struct {
+	mappings map[string][]string
+}
+
+func (m *GroupMapper) AllGroupIDs(ctx context.Context) ([]string, error) {
+	res := make([]string, 0)
+	for key := range m.mappings {
+		res = append(res, key)
+	}
+	slices.Sort(res)
+	return res, nil
+}
+
+func (m GroupMapper) ContainsGroupID(ctx context.Context, key string) (bool, error) {
+	_, ok := m.mappings[key]
+	if !ok {
+		return false, fmt.Errorf("group %s is not mapped", key)
+	}
+	return ok, nil
+}
+
+func (m GroupMapper) MappedGroupIDs(ctx context.Context, key string) ([]string, error) {
+	x, ok := m.mappings[key]
+	if !ok {
+		return nil, fmt.Errorf("no mapping found for group ID: %s", key)
+	}
+	return x, nil
+}
+
+type GoogleGroupToGitHubMapper GroupMapper
+
+type GitHubToGoogleGroupMapper GroupMapper
```

Comment:
I'm confused by the goal here.

- Are we trying to make `GoogleGroupToGitHubMapper` and `GitHubToGoogleGroupMapper` be interchangeable? In that case, we should use an interface and have each struct implement an interface.

- Are we trying to share functionality about set operations on the mapper? In that case, we should still use an interface, but have these map operations accept the interface (or output of the interface). Example:

```go
type GroupMapper interface {
  // Mappings returns a KV mapping of email addresses to lists of group/team identities in the target system.
  Mappings() map[string][]string
}
```

Then you can have a `maps` package that accepts the `GroupMapper` interface _or_ just operates on `map[string][]string` inputs.